src/services/tyeProcessProvider.ts

```diff
@@ -60,26 +79,48 @@ export default class LocalTyeProcessProvider implements TyeProcessProvider {
         return this.processes.pipe(first()).toPromise();
     }
 
-    private async getProcessList(): Promise<TyeProcess[]> {
+    private async getTyeProcessesWithPorts(): Promise<TyeProcessWithPorts[]> {
         const tyePath = await this.tyePathProvider.getTyePath();
         const tyeProcesses = await this.processProvider.listProcesses(tyePath);
-        const tyeProcessesWithPorts = await Promise.all(tyeProcesses.map(process => this.getPortForProcess(process.pid)));
+        
+        return await Promise.all(tyeProcesses.map(process => this.getTyeProcessWithPorts(process.pid)));
+    }
 
-        function hasValidPort(process: ProspectiveTyeProcess): process is TyeProcess {
-            return !!process.dashboardPort;
-        }
+    private async getTyeProcessesFromPorts(processes: TyeProcessWithPorts[]): Promise<TyeProcess[]> {
+        const allProcesses = await Promise.all(processes.map(process => this.getTyeProcessFromPorts(process)));
 
-        const tyeProcessesWithValidPorts =
-            tyeProcessesWithPorts
-                .filter(hasValidPort);
+        function isValidProcess(process: TyeProcess | undefined): process is TyeProcess {
+            return !!process;
+        }
 
-        return tyeProcessesWithValidPorts;
+        return allProcesses.filter(isValidProcess);
     }
 
-    private async getPortForProcess(pid: number): Promise<ProspectiveTyeProcess> {
+    private async getTyeProcessWithPorts(pid: number): Promise<TyeProcessWithPorts> {
         const ports = await this.portProvider.getPortsForProcess(pid);
 
-        return { pid, dashboardPort: ports[0] }
+        return { pid, ports };
     }
-}
 
+    private async getTyeProcessFromPorts(process: TyeProcessWithPorts): Promise<TyeProcess | undefined> {
+        for (const port of process.ports) {
+            const dashboardUri = vscode.Uri.parse(`http://localhost:${port}`);
+            const client = this.tyeClientProvider(dashboardUri);
+
+            if (client) {
+                try {
+                    const endpoints = await client.getEndpoints();
+
+                    if (endpoints) {
```

Comment:
Consider adding a comment that better verification of the dashboard endpoint is planned. In a microservice app there might be many things that respond to `api/v1` path ☺️

Reply:
Yes, perhaps we could add a more "uniquely Tye" endpoint to the dashboard.  (I need to dust off my mDNS PR for Tye, which would sidestep the issue in that Tye itself could advertise its dashboard port, but haven't had time.)

Reply:
Added TODO.